### PR TITLE
TMDM-11185 Improve REST client error handling

### PR DESCRIPTION
**Pull request by qcui**
https://jira.talendforge.org/browse/TMDM-11185

What is the current behavior? 
When issuing REST calls, there's no distinctions between 401, 403 errors from other errors, then a generic error message is returned

What is the new behavior?
Improve error handling by returning specific error messages when you're not authorized or you do not have enough priviledges to access an endpoint.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by qcui** [comment]
The 403 response status should defined by GWT, MDM didn't return any status information to client when  authority validate failed.
For display message FR translation need you provide detail.

File: org.talend.mdm.webapp.base/src/main/resources/org/talend/mdm/webapp/base/client/i18n/BaseMessages_fr.properties [hunk 1]
```diff
@@ -36,6 +36,7 @@ restore_confirm=Etes-vous s\u00fbr de vouloir restaurer cet enregistrement ?
 
 server_error=Erreur serveur
 server_error_notification=Une erreur s''est produite. Veuillez consulter le log serveur pour plus de d\u00e9tails.
+server_error_forbidden_resource=Resource access forbidden.
```
Comment (pbabin) [inline org.talend.mdm.webapp.base/src/main/resources/org/talend/mdm/webapp/base/client/i18n/BaseMessages_fr.properties]: `server_error_forbidden_resource=L''acc\u00e8s \u00e0 la ressource est refus\u00e9`

File: org.talend.mdm.webapp.base/src/main/resources/org/talend/mdm/webapp/base/client/i18n/BaseMessages_en.properties [hunk 2]
```diff
@@ -37,6 +37,7 @@ overwrite_confirm=This record conflicts with an existing record of same identifi
 
 server_error=Server Error
 server_error_notification=Error occured. See server log for details.
+server_error_forbidden_resource=Resource access forbidden.
```
Comment (pbabin) [inline org.talend.mdm.webapp.base/src/main/resources/org/talend/mdm/webapp/base/client/i18n/BaseMessages_en.properties]: `server_error_forbidden_resource=Access to resource is denied`

File: org.talend.mdm.webapp.base/src/main/java/org/talend/mdm/webapp/base/client/rest/ClientResourceWrapper.java [hunk 3]
```diff
@@ -74,6 +74,9 @@ public void handle(Request request, Response response) {
                 // http://stackoverflow.com/questions/10046972/msie-returns-status-code-of-1223-for-ajax-request
                 if (statusCode >= 200 && statusCode <= 299 || statusCode == 1223) {
                     callbackHandler.process(request, response);
+                } else if (statusCode == 403) {
```
Comment (pbabin) [inline org.talend.mdm.webapp.base/src/main/java/org/talend/mdm/webapp/base/client/rest/ClientResourceWrapper.java]: Add a case for 401
`server_error_unauthorized=Unauthorized access`
`server_error_unauthorized=Acc\u00e8s non autoris\u00e9`

Comment (qcui) [reply org.talend.mdm.webapp.base/src/main/java/org/talend/mdm/webapp/base/client/rest/ClientResourceWrapper.java]: ok, I will add it.

**Comment by qcui** [comment]
updated code, please review.

File: org.talend.mdm.webapp.base/src/main/java/org/talend/mdm/webapp/base/client/rest/ClientResourceWrapper.java [hunk 4]
```diff
@@ -74,6 +74,12 @@ public void handle(Request request, Response response) {
                 // http://stackoverflow.com/questions/10046972/msie-returns-status-code-of-1223-for-ajax-request
                 if (statusCode >= 200 && statusCode <= 299 || statusCode == 1223) {
                     callbackHandler.process(request, response);
+                } else if (statusCode == 401) {
+                    MessageBox.alert(BaseMessagesFactory.getMessages().server_error(), BaseMessagesFactory.getMessages()
+                            .server_error_unauthorized(), null);
+                } else if (statusCode == 403) {
```
Comment (htyin) [inline org.talend.mdm.webapp.base/src/main/java/org/talend/mdm/webapp/base/client/rest/ClientResourceWrapper.java]: There are constants defined for 401 and 403 in HttpStatus. It is more clear for the meaning of status code.
HttpStatus.SC_UNAUTHORIZED 
HttpStatus.SC_FORBIDDEN

**Comment by qcui** [comment]
HttpStatus needs import httpcomponents jar and HttpStatus has no 1223 status.

**Comment by qcui** [comment]
change updateLastSyncTime function

**Pull request merged**